partiql-planner/src/main/kotlin/org/partiql/planner/PartiQLHeader.kt

```diff
@@ -347,7 +347,7 @@ public object PartiQLHeader : Header() {
         )
     }
 
-    private fun between(): List<FunctionSignature.Scalar> = types.numeric.map { t ->
+    private fun between(): List<FunctionSignature.Scalar> = (types.numeric + types.text).map { t ->
```

Comment:
```suggestion
    private fun between(): List<FunctionSignature.Scalar> = (types.numeric + types.text + BOOL).map { t ->
```

Given the other changes, should we add this?

Reply:
I choose to not add this in this PR because I hope once we have finalized missing prorogation rule, we can just perform a rewrite on Between. 

`Bool BETWEEN Bool AND Bool` on itself does not make much sense to me, and at the moment this is no use case require this function.